docs/src/components/Layout/Sidebar.tsx

```diff
@@ -0,0 +1,290 @@
+  }
+
+  return (
+    <Link href={`/${platform}${href}`} passHref>
+      <a onClick={onClick} className={classNames}>
+        {children}
+      </a>
+    </Link>
+  );
+};
+
+const NavLinkComponentsSection = ({ heading, components, ...props }) => {
+  const { query } = useCustomRouter();
+  const { tokens } = useTheme();
+  const { platform = 'react' } = query;
+
+  const platformComponents = components.filter((component) => {
+    if (component.platforms) {
+      return component.platforms.includes(platform);
+    }
+    return true;
+  });
+
+  if (!platformComponents.length) {
+    return null;
+  }
+  return (
+    <>
+      <Text
+        fontSize={tokens.fontSizes.xs}
+        fontWeight={tokens.fontWeights.semibold}
+        textTransform="uppercase"
+        letterSpacing="0.125em"
+        color={tokens.colors.font.tertiary}
+        padding={`${tokens.space.small} ${tokens.space.medium} ${tokens.space.xs} var(--secondary-nav-indent)`}
+      >
+        {heading}
+      </Text>
+      <NavLinks {...props} items={platformComponents} />
+    </>
+  );
+};
+
+const ExpanderTitle = ({ Icon, text }) => {
+  const { tokens } = useTheme();
+  return (
+    <Flex
+      direction="row"
+      alignItems="center"
+      gap={tokens.space.small}
+      padding={tokens.space.xs}
+    >
+      <Icon
+        style={{ height: '1.25rem', width: '1.25rem' }}
+        color={tokens.colors.brand.primary[60]}
+      />
+      <Text as="span">{text}</Text>
+    </Flex>
+  );
+};
+
+// TODO: clean up this logic
+const SecondaryNav = (props) => {
+  const {
+    pathname,
+    query: { platform },
+  } = useCustomRouter();
+
+  // Extract section from URL (/section/... => section)
+  const section = pathname.split('/')[2];
+  const [value, setValue] = React.useState<string | string[]>([section]);
+
+  return (
+    <Expander type="multiple" value={value} onChange={setValue}>
+      <ExpanderItem
+        title={
+          <ExpanderTitle Icon={MdOutlineChecklist} text="Getting started" />
+        }
+        value="getting-started"
+      >
+        {gettingStarted.map(({ label, ...rest }) => (
+          <NavLink key={label} {...rest} onClick={props.onClick}>
+            {label}
+          </NavLink>
+        ))}
+      </ExpanderItem>
+      <ExpanderItem
+        title={
+          <ExpanderTitle Icon={MdOutlinePower} text="Connected components" />
+        }
+        value="components"
+      >
+        {connectedComponents.map(({ label, ...rest }) => (
+          <NavLink key={label} {...rest} onClick={props.onClick}>
+            {label}
+          </NavLink>
+        ))}
+      </ExpanderItem>
+      <ExpanderItem
+        title={
+          <ExpanderTitle Icon={MdOutlineWidgets} text="Primitive components" />
+        }
+        value="components"
+      >
+        <>
+          <NavLinkComponentsSection
+            {...props}
+            heading={'Base'}
+            components={baseComponents}
+          ></NavLinkComponentsSection>
```

Comment:
Consider creating an array using the `heading` and `components` values and mapping over it for easier addition/removal in the future e.g.
```
const navLinkSections = [{ heading: 'Base', components: baseComponents }, heading: 'Feedback', components: feedbackComponents];
....
<>
  {navLinkSections.map(({ heading, components }) => (
    <NavLinkComponentsSection {...props} heading={heading} components={components} />
  )}
</>
```

Reply:
ah yea that will clean this up a bunch